vowpalwabbit/core/include/vw/core/metrics_manager.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.=
+
+#pragma once
+
+#include "metric_sink.h"
+#include "vw/core/learner_fwd.h"
+
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace VW
+{
+class metrics_manager
+{
+public:
+  metrics_manager(bool enabled = false, std::string filename = "");
+
+  using metrics_callback_fn = std::function<void(VW::metric_sink&)>;
+
+  bool are_metrics_enabled() const;
+  std::string get_filename() const;
```

Comment:
get_filename should not be here if the class never uses it, it should probably still live in metrics, or are there plans for it to be used somehow in the future?

Reply:
reverted back to `std::string filename = all.options->get_typed_option<std::string>("extra_metrics").value();`